### ISSUE TITLE
SVG with linked or embedded images are not rendered correctly

### ISSUE THREAD

**Issue by gern0** [asciidoctor/asciidoctor-pdf#1732]
Image elemets (either linked or embedded) in SVGs used as "title page background" will not be rendered.
For example: I have a SVG with an embedded (or linked) PNG as background image also including other forms (e.g. rectangles).
If such a SVG is used as title_page_background_image only the forms (e.g. rectangles) get rendered but not the embedded or linked background image.

**Comment by mojavelinux** [comment]
It's likely that the referenced image cannot be found. There is a test that covers this scenario in the test suite. See https://github.com/asciidoctor/asciidoctor-pdf/blob/4075c627bc27b6f1d9cd94cd6bddee88aff50461/spec/page_spec.rb#L882-L890.

Please provide a complete example so that I can reproduce it. Then I'll be able to advise you about whether it's a configuration problem or a defect in the software.

**Comment by gern0** [comment]
Thanks for the fast answer. At first I couldn't reproduce the problem with an example content.
It seems that the problem is occuring because of the internal structure of the SVG.
The dysfunctional way got an SVG exported by Affinity Publisher like:
```
<svg ...>
    <g clip-path="url(#_clip1)">
        <g id="Crypt_35" transform="matrix(0.24,0,0,0.24,-297.6,-420.96)">
            <use xlink:href="#_Image2" x="0" y="0" width="2480px" height="3508px"/>
        </g>
    </g>

     ... other forms ...

    <defs>
        <image id="_Image2" width="2480px" height="3508px" xlink:href="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQEAY ... />
    </defs>
</svg>
```

Other applications (e.g. eog, inkscape, gimp) had no problems with defining the image object in the defs section after the reference in `<use xlink:href="#_Image2"`.

After manually putting the def-section before the reference everything got rendered correctly.

**Comment by mojavelinux** [comment]
I'm glad to hear you worked it out.

Asciidoctor PDF does not, itself, interpret the SVG and embed the objects into the PDF. That's handled by prawn-svg. I recommend reporting this issue there. See https://github.com/mogest/prawn-svg

**Comment by gern0** [comment]
Thanks for your answer and your amazing work.

**Comment by mojavelinux** [comment]
Thanks!

I'll close this and mark it as upstream.

**Comment by mojavelinux** [comment]
Upstream has been released with this feature. I've upgraded the master branch to use it (to be included in the 2.0 release).

One thing I noticed is that the mimetype "data:image/jpeg" is not supported. It must be "data:image/jpg".
